### ISSUE TITLE
Serving Halcyon via Caddy

### ISSUE THREAD

**Issue by mcrosson** [halcyon-suite/halcyon#14]
Just a quick heads up / note. I was able to get Halcyon working with Caddy. I've included a simple config below in case anyone is looking for an example.

```
halcyon.domain.tld  {
  tls user@domain.tld

  log /var/log/caddy/halcyon.log

  fastcgi / 127.0.0.1:9000 php

  root /opt/halcyon

status 404 {
/config.ini
/.bat
/.git
/.ini
/.sh
/.svn
/.txt
/.tpl
/.xml
}

rewrite {
  r ^/home/?$
  to /
}
rewrite {
  r ^/login/?$
  to /login/login.php
}
rewrite {
  r ^/auth/?$
  to /login/auth.php
}
rewrite {
  r ^/logout/?$
  to /login/logout.php
}
rewrite {
  r ^/terms/?$
  to /login/terms.php
}
rewrite {
  r ^/local/?$
  to /local.php
}
rewrite {
  r ^/federated/?$
  to /federated.php
}
rewrite {
  r ^/notifications/?$
  to /notifications.php
}
rewrite {
  r ^/search/?$
  to /search_hash_tag.php
}
rewrite {
  r ^/search/users/?$
  to /search_user.php
}
rewrite {
  r ^/@(.+)@(.+).([a-z]+)/?$
  to /user.php?user=@{1}@{2}.{3}
}
rewrite {
  r ^/@(.+)@(.+).([a-z]+)/status/(.+?)?$
  to /user.php?user=@{1}@{2}.{3}&status={4}
}
rewrite {
  r ^/@(.+)@(.+).([a-z]+)/media/?$
  to /user_only_media.php?user=@{1}@{2}.{3}
}
rewrite {
  r ^/@(.+)@(.+).([a-z]+)/with_replies/?$
  to /user_include_replies.php?user=@{1}@{2}.{3}
}
rewrite {
  r ^/@(.+)@(.+).([a-z]+)/followers/?$
  to /user_followers.php?user=@{1}@{2}.{3}
}
rewrite {
  r ^/@(.+)@(.+).([a-z]+)/following/?$
  to /user_following.php?user=@{1}@{2}.{3}
}
rewrite {
  r ^/@(.+)@(.+).([a-z]+)/favourites/?$
  to /user_favorite.php?user=@{1}@{2}.{3}
}
rewrite {
  r ^/avatars/original/missing.png$
  to /assets/images/missing.png
}
rewrite {
  r ^/headers/original/missing.png$
  to /assets/images/missing_header.png
}
rewrite {
  r ^/404/?$
  to /404.php
}
}
```

**Comment by nipos** [comment]
Thank you very much for that example.
I'll include that example config in the next release of Halcyon.
Caddy is now officially supported!

**Comment by nipos** [comment]
The Caddyfile is now included in Halcyon 1.2.0.
Please note that the new version requires some additional rewrites.
You should use my modified file on your server to make 1.2.0 work correctly.
https://notabug.org/halcyon-suite/halcyon/src/master/Caddyfile
